handlers/officerDeltaHandler.go

```diff
@@ -28,34 +28,39 @@ func NewOfficerDeltaHandler(kSvc services.KafkaService, h helpers.Helper, chv va
 // encountered then they will be returned via the ResponseWriter.
 func (kp *OfficerDeltaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
-	log.Info(fmt.Sprintf("Open API spec to use: %s", kp.cfg.OpenApiSpec), nil)
+	contextId, err := kp.h.GetRequestIdFromHeader(r)
```

Comment:
Instead of doing this in the handler could you include this in the chOpenApiValidator.go code. We pass the request object there . That. way it will only be done in the common code and won't need to repeat this in all the handlers.

Reply:
I would suggest keeping it in the handler as otherwise KafkaService will need to have a helper struct attached too it which would be passed in anyway in place of the contextId